doc/widget_index.py

```diff
@@ -0,0 +1,46 @@
+"""
+Generete widget index for the webpage.
+"""
+
+import os
+from bs4 import BeautifulSoup
```

Comment:
Can you group imports: https://www.python.org/dev/peps/pep-0008/#imports
It makes it easier to see which external libraries are needed when their imports are not lost between python std library imports.